Client/Frontend/Settings/CustomSearchViewController.swift

```diff
@@ -0,0 +1,206 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import Shared
+import SnapKit
+import Storage
+import WebImage
+import Deferred
+
+private let log = Logger.browserLogger
+
+class CustomSearchViewController: SettingsTableViewController {
+    
+    private var urlString: String?
+    private var engineTitle = ""
+    private var spinnerView = UIActivityIndicatorView(activityIndicatorStyle: .Gray)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Add Custom Search Engine"
```

Comment:
This will need to be localized so we should wrap it in the `NSLocalizedString` function and place it in `Strings.swift`